tensorpipe/core/message.h

```diff
@@ -49,9 +51,7 @@ class Message final {
   std::vector<Payload> payloads;
 
   struct Tensor {
-    void* data{nullptr};
-    size_t length{0};
-
+    tensorpipe::Tensor data;
```

Comment:
I think that here we should keep both the newly-introduced `data` field but, for now, to preserve backwards compatibility, we could also keep the previous "raw" fields (`data` and `length`) and have the pipe convert these raw fields to a CpuTensor and then use that going forward. This would allow us to land this change, migrate PyTorch over to use the new field, and then drop the old ones.

Unfortunately this means that we need to find another name for the new `data` field, as otherwise it conflicts with the old one...